## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 428442`
Commit: `0a1c3249a287685384b066b2b1cc47a31bfafd7d`

---
### `x86_64-darwin`
<details>
  <summary>:x: 3 packages failed to build:</summary>
  <ul>
    <li>tailscale</li>
    <li>tailscale.derper</li>
    <li>trayscale</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>tailscale-gitops-pusher</li>
    <li>tailscale-nginx-auth</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-darwin`</summary>
<details>
<summary>tailscale</summary>
<pre>ok  	tailscale.com/tempfork/acme	4.335s
ok  	tailscale.com/tempfork/gliderlabs/ssh	0.013s
ok  	tailscale.com/tempfork/heap	0.011s
ok  	tailscale.com/tempfork/sshtest/ssh	0.879s
ok  	tailscale.com/tka	1.342s
ok  	tailscale.com/tsconsensus	46.899s
ok  	tailscale.com/tsnet	0.024s [no tests to run]
ok  	tailscale.com/tstest	0.033s
ok  	tailscale.com/tstest/archtest	0.008s
ok  	tailscale.com/tstest/deptest	0.015s
ok  	tailscale.com/tstest/iosdeps	0.315s
ok  	tailscale.com/tstest/jsdeps	0.297s
ok  	tailscale.com/tstest/natlab	0.013s
--- FAIL: TestProtocolQEMU (0.00s)
    vnet_test.go:539: listen unix /nix/var/nix/builds/nix-build-tailscale-1.86.2.drv-1289-2658856026/TestProtocolQEMU331567467/001/vnet.sock: bind: invalid argument
--- FAIL: TestProtocolUnixDgram (0.00s)
    vnet_test.go:588: listen unixgram /nix/var/nix/builds/nix-build-tailscale-1.86.2.drv-1289-2658856026/TestProtocolUnixDgram3125167875/001/vnet.sock: bind: invalid argument
FAIL
FAIL	tailscale.com/tstest/natlab/vnet	0.053s
FAIL</pre>
</details>
</details>

--- a/pkgs/by-name/ta/tailscale/package.nix
+++ b/pkgs/by-name/ta/tailscale/package.nix
@@ -23,7 +23,7 @@
 }:
 
 let
-  version = "1.84.3";
+  version = "1.86.2";
 in
 buildGoModule {
   pname = "tailscale";
@@ -37,11 +37,11 @@ buildGoModule {
   src = fetchFromGitHub {
     owner = "tailscale";
     repo = "tailscale";
-    rev = "v${version}";
-    hash = "sha256-0HvUNpyi6xzS3PtbgMvh6bLRhV77CZRrVSKGMr7JtbE=";
+    tag = "v${version}";
+    hash = "sha256-hozfvKkvTeaabN1tYl0NlEpjfD4sZQe9Z+agdoXFHNE=";
   };
 
-  vendorHash = "sha256-QBYCMOWQOBCt+69NtJtluhTZIOiBWcQ78M9Gbki6bN0=";
+  vendorHash = "sha256-4QTSspHLYJfzlontQ7msXyOB5gzq7ZwSvWmKuYY5klA=";
 
   nativeBuildInputs = [
     makeWrapper
@@ -63,8 +63,7 @@ buildGoModule {
   ];
 
   excludedPackages = [
-    # exlude integration tests which fail to work
-    # and require additional tooling
+    # Exclude integration tests which fail to work and require additional tooling
     "tstest/integration"
   ];
 
@@ -79,8 +78,7 @@ buildGoModule {
     "ts_include_cli"
   ];
 
-  # remove vendored tooling to ensure it's not used
-  # also avoids some unnecessary tests
+  # Remove vendored tooling to ensure it's not used; also avoids some unnecessary tests
   preBuild = ''
     rm -rf ./tool
   '';
@@ -151,6 +149,12 @@ buildGoModule {
 
         # flaky: https://github.com/tailscale/tailscale/issues/15348
         "TestSafeFuncHappyPath"
+
+        # Requires `go` to be installed with the `go tool` system which we don't use
+        "TestGoVersion"
+
+        # Fails because we vendor dependencies
+        "TestLicenseHeaders"
       ]
       ++ lib.optionals stdenv.hostPlatform.isDarwin [
         # syscall default route interface en0 differs from netstat
@@ -165,6 +169,11 @@ buildGoModule {
 
         # Fails only on Darwin, succeeds on other tested platforms.
         "TestOnTailnetDefaultAutoUpdate"
+
+        # Fails due to UNIX domain socket path limits in the Nix build environment.
+        # Likely we could do something to make the paths shorter.
+        "TestProtocolQEMU"
+        "TestProtocolUnixDgram"
       ];
     in
     [ "-skip=^${builtins.concatStringsSep "$|^" skippedTests}$" ];

